From: Joan Torres Lopez <joantolo@redhat.com>
Date: Thu, 29 Jan 2026 14:04:40 +0100
Subject: inputMethod: Honor the HIDDEN_TEXT hint of IBusInputHint

Makes the pre-edit text hidden with a "_".

This hint was added in IBus 1.5.34, although if IBus doesn't support it,
it will silently fallback to no-op which is fine.

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/4053>
(cherry picked from commit a39035fda4e1cf487b83483e2d4fe41a830d7fea)
Origin: upstream, 49.4, commit:9558a6cc80cf4c5b187cd77c220acedc70902dff
---
 js/misc/inputMethod.js | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/js/misc/inputMethod.js b/js/misc/inputMethod.js
index ee77166..caa1067 100644
--- a/js/misc/inputMethod.js
+++ b/js/misc/inputMethod.js
@@ -292,6 +292,8 @@ export const InputMethod = GObject.registerClass({
             ibusHints |= IBus.InputHints.UPPERCASE_WORDS;
         if (hints & Clutter.InputContentHintFlags.SENSITIVE_DATA)
             ibusHints |= IBus.InputHints.PRIVATE;
+        if (hints & Clutter.InputContentHintFlags.HIDDEN_TEXT)
+            ibusHints |= IBus.InputHints.HIDDEN_TEXT;
 
         this._hints = ibusHints;
         if (this._context)
